src/olympia/blocklist/tests/test_tasks.py

```diff
@@ -157,53 +172,185 @@ def _block_version(
 
         assert self.mocks['complete_session'].called
-        assert self.mocks['set_config'].call_args_list == [
-            mock.call(MLBF_TIME_CONFIG_KEY, self.generation_time, json_value=True),
-            mock.call(MLBF_BASE_ID_CONFIG_KEY, self.generation_time, json_value=True),
-        ]
+        assert (
+            mock.call(MLBF_TIME_CONFIG_KEY, self.generation_time, json_value=True)
+        ) in self.mocks['set_config'].call_args_list
+
+        for block_type in block_types:
+            if block_type == BlockType.BLOCKED:
+                # We currently write to the old singular config key for hard blocks
+                # to preserve backward compatibility.
+                # In https://github.com/mozilla/addons/issues/15193
+                # we can remove this and start writing to the new plural key.
+                assert (
+                    mock.call(
+                        MLBF_BASE_ID_CONFIG_KEY(block_type, compat=True),
+                        self.generation_time,
+                        json_value=True,
+                    )
+                ) in self.mocks['set_config'].call_args_list
+            assert (
+                mock.call(
+                    MLBF_BASE_ID_CONFIG_KEY(block_type),
+                    self.generation_time,
+                    json_value=True,
+                )
+            ) in self.mocks['set_config'].call_args_list
+
+    def test_upload_blocked_filter(self):
+        self._test_upload_base_filter(BlockType.BLOCKED)
+
+    def test_upload_soft_blocked_filter(self):
+        self._test_upload_base_filter(BlockType.SOFT_BLOCKED)
+
+    def test_upload_soft_and_blocked_filter(self):
+        self._test_upload_base_filter(BlockType.BLOCKED, BlockType.SOFT_BLOCKED)
+
+    def _test_cleanup_old_records(
+        self,
+        filter_list: Dict[BlockType, int],
+        records: List[Dict[str, int]],
+        expected_calls: List[any],
+    ):
+        self._block_version(is_signed=True)
+        mlbf = MLBF.generate_from_db(self.generation_time)
+        for block_type, base_id in filter_list.items():
+            mlbf.generate_and_write_filter(block_type)
+            # We currently write to the old singular config key for hard blocks
+            # to preserve backward compatibility.
+            # In https://github.com/mozilla/addons/issues/15193
+            # we can remove this and start writing to the new plural key.
+            set_config(
+                MLBF_BASE_ID_CONFIG_KEY(block_type, compat=True),
+                base_id,
+                json_value=True,
+            )
 
-    def test_upload_stashed_filter(self):
+        self.mocks['records'].return_value = records
+        upload_filter.delay(
+            self.generation_time,
+            filter_list=[block_type.name for block_type in filter_list],
+        )
+
+        assert self.mocks['delete_record'].call_args_list == expected_calls
```

Comment:
We'll want to make sure that `delete_session` is called too.

Reply:
you mean `complete_session` right?

Reply:
oops correct

Reply:
I have a test for that specifically.